app/src/main/java/de/danoeh/antennapod/dialog/SwipeActionsDialog.java

```diff
@@ -0,0 +1,187 @@
+package de.danoeh.antennapod.dialog;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.graphics.PorterDuff;
+import android.graphics.drawable.Drawable;
+import android.view.LayoutInflater;
+import android.view.View;
+import androidx.appcompat.app.AlertDialog;
+import androidx.appcompat.content.res.AppCompatResources;
+import androidx.core.content.ContextCompat;
+import androidx.core.graphics.drawable.DrawableCompat;
+import androidx.gridlayout.widget.GridLayout;
+
+import com.annimon.stream.Stream;
+
+import de.danoeh.antennapod.R;
+import de.danoeh.antennapod.databinding.FeeditemlistItemBinding;
+import de.danoeh.antennapod.databinding.SwipeactionsDialogBinding;
+import de.danoeh.antennapod.databinding.SwipeactionsPickerBinding;
+import de.danoeh.antennapod.databinding.SwipeactionsPickerItemBinding;
+import de.danoeh.antennapod.databinding.SwipeactionsRowBinding;
+import de.danoeh.antennapod.fragment.EpisodesFragment;
+import de.danoeh.antennapod.fragment.FeedItemlistFragment;
+import de.danoeh.antennapod.fragment.QueueFragment;
+import de.danoeh.antennapod.fragment.swipeactions.SwipeAction;
+import de.danoeh.antennapod.fragment.swipeactions.SwipeActions;
+import de.danoeh.antennapod.ui.common.ThemeUtils;
+
+import java.util.List;
+
+public class SwipeActionsDialog {
+    private static final int LEFT = 1;
+    private static final int RIGHT = 0;
+
+    private final Context context;
+    private final String tag;
+
+    private SwipeAction rightAction;
+    private SwipeAction leftAction;
+    private List<SwipeAction> keys;
+
+    public SwipeActionsDialog(Context context, String tag) {
+        this.context = context;
+        this.tag = tag;
+    }
+
+    public void show(Callback prefsChanged) {
+        List<SwipeAction> rightLeft = SwipeActions.getPrefsWithDefaults(context, tag);
```

Comment:
I think it's a bit strange to use a list here. How about using a Pair from `androidx.core.util`? Makes it more clear that we can always expect 2 items.

Reply:
sure